File: Module.Persist.TPM/Model/History/HistoricalMechanic.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(Mechanic))]
    public class HistoricalMechanic : BaseHistoricalEntity<System.Guid>
    {
        public string Name { get; set; }
        public string SystemName { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/mechanic/HistoricalMechanic.js
﻿Ext.define('App.model.tpm.mechanic.HistoricalMechanic', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'Mechanic',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true },
        { name: 'SystemName', useNull: true, type: 'string', hidden: false, isDefault: true }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalMechanics',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
